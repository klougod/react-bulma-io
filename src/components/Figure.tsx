import React from 'react'
import PropTypes from 'prop-types'

export const Figure = ({ className, alt, src }: any) => (
  <figure className={`image ${className}`}>
    <img src={src} alt={alt} />
  </figure>
)

Figure.defaultProps = {
  className: '',
  src: 'https://bulma.io/images/placeholders/128x128.png',
  alt: 'image does not have alt'
}

Figure.propTypes = {
  className: PropTypes.string,
  src: PropTypes.string,
  alt: PropTypes.string
}

export default Figure
