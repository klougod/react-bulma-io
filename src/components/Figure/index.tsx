import React, { ImgHTMLAttributes } from 'react'

export const Figure = ({ className, ...rest}: ImgHTMLAttributes<HTMLImageElement>) => (
  <figure className={`image ${className}`}>
    <img {...rest} />
  </figure>
)

Figure.defaultProps = {
  className: '',
  src: 'https://bulma.io/images/placeholders/128x128.png',
  alt: 'image does not have alt'
}

export default Figure
