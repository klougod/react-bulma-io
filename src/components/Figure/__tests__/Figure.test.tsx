import React from 'react'
import Enzyme from 'enzyme'

import Figure from '../'

describe('Figure component test', function () {
  it('should render without crash without props', function () {
    const figureMock = Enzyme.shallow(<Figure />)
    expect(figureMock).toMatchSnapshot()
  })

  it('should render without crash with props', function () {
    const figureMockProps = {
      className: 'figureCLass',
      src: 'imageSrcUrl',
      alt: 'image alt'
    }
    const figureMock = Enzyme.shallow(<Figure {...figureMockProps} />)
    expect(figureMock).toMatchSnapshot()
  })
})
