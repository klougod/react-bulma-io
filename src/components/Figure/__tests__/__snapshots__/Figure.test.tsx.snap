// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Figure component test should render without crash with props 1`] = `
<figure
  className="image figureCLass"
>
  <img
    alt="image alt"
    src="imageSrcUrl"
  />
</figure>
`;

exports[`Figure component test should render without crash without props 1`] = `
<figure
  className="image "
>
  <img
    alt="image does not have alt"
    src="https://bulma.io/images/placeholders/128x128.png"
  />
</figure>
`;
